plugins/CoreHome/javascripts/dataTable.js

```diff
@@ -720,6 +720,8 @@ $.extend(DataTable.prototype, UIControl.prototype, {
                 return;
             }
 
+            $('.periodName', domElem).html(period);
```

Comment:
Guess it would be needed to convert this to a translated text:
```suggestion
            var newPeriod = piwikPeriods.get(period);
            $('.periodName', domElem).html(newPeriod.getDisplayText());
```